feat: Add login and registration functionality.

Melany0802 opened the pull request.

Melany0802 committed:
feat: Add login and registration functionality.

zSnails commented on app/src/main/java/tec/ac/cr/marape/app/LoginActivity.kt:
```diff
@@ -1,11 +1,92 @@
+      //Verificar Usuario
+      mAuth.signInWithEmailAndPassword(email, contrasenia)
+        .addOnCompleteListener { task ->
+          if (task.isSuccessful) {
```
Aquí falta que setee el usuario que inició sesión en el state manager global

zSnails commented on app/src/main/java/tec/ac/cr/marape/app/RegisterActivity.kt:
```diff
@@ -1,11 +1,90 @@
+              //Redireccionar al MainActivity
+              val principal = Intent(this, MainActivity::class.java)
+              //Iniciar la activity
+              startActivity(principal)
```
Aquí también, antes de lanzar la actividad tienes que settear el usuario que inició sesión en el state manager global

zSnails commented on app/src/main/java/tec/ac/cr/marape/app/RegisterActivity.kt:
```diff
@@ -1,11 +1,90 @@
+      var password = passwordEntry.text.toString()
+      var confirmPassword = confirmPasswordEntry.text.toString()
+
+      if (userName.isEmpty() || userName.length < 5){
```
Cadena de if else if else se podría re-escribir como un `when` statement para estas validaciones.

zSnails commented on app/src/main/java/tec/ac/cr/marape/app/LoginActivity.kt:
```diff
@@ -1,11 +1,92 @@
 package tec.ac.cr.marape.app
```
Darle formato al archivo.

zSnails reviewed (changes requested):
Falta darle formato a todo

Melany0802 committed:
refactor: Solve requested changes

Melany0802 committed:
refactor: Change usage of the onComplete callback and instead opt into using onSuccess and onFailure

zSnails committed:
fix: Automatically load the user upon launching the app if the user is already logged in

zSnails committed:
fix: Add account deletion confirmation dialog.

For some reason there wasn't a confirmation dialog, and we really can't afford to not have one, a simple click
on the button would have deleted that user's account, although inventories aren't automatically deleted when
the account is deleted so that would've been fine

zSnails committed:
fix: Remove some of the old hard-coded strings and put them into the strings resource file.

zSnails committed:
fix: Re make the login activity's layout so that it's now responsive and properly done.

The old layout was fine but it was prone to errors and positioning artifacts, now I've fixed that.

zSnails committed:
Merge branch 'main' into impl-login-activity
